src/Plugin/RulesExpression/ActionSet.php

```diff
@@ -77,7 +89,7 @@ public function addExpressionObject(ExpressionInterface $expression) {
     if (!$expression instanceof ActionExpressionInterface) {
       throw new InvalidExpressionException();
     }
-    $this->actions[] = $expression;
+    $this->actions[$this->uuidService->generate()] = $expression;
```

Comment:
Maybe we could make the manager responsible of generating IDs for expressions ? It seems a bit weird that every container has to take care of that. Likewise, I think we need a "global UUID lookup", which works at least on the configuration level as we do not know the container something is contaiend in either?